src/Angular.js

```diff
@@ -125,6 +128,39 @@ var VALIDITY_STATE_PROPERTY = 'validity';
 
 var hasOwnProperty = Object.prototype.hasOwnProperty;
 
+var objectMaxDepthInErrorMessage = 5;
+
+/**
+ * @ngdoc function
+ * @name angular.configureErrorHandling
+ * @module ng
+ * @kind function
+ *
+ * @description Use this function to configure error handling.
+ *
+ * @param {Object=} config An object for defining configuration for error handling. The
+ *     following keys are supported:
+ *
+ * * `objectMaxDepth` - used to stringify objects in error messages until reaching the max depth.
+ * if it is not supplied the default max depth which is 5 will be used instead, or if it is invalid
+ * (less than `1` or not a `Number`) no max depth will be used.
+ *
+ */
+function configureErrorHandling(config) {
+  if (isObject(config)) {
+    objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;
+  }
+}
```

Comment:
@gkalpak 
I have changed the implementation to
 
``` javascript
function errorHandlingConfig(config) {
  if (!isObject(config)) {
    return {
      objectMaxDepth: objectMaxDepthInErrorMessage
    };
  } else {
    if (isDefined(config.objectMaxDepth)) {
      objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;
    }
  }
}
``` 
so it will work as a setter when config (object type ) is supplied.
please tell me if there anything missing.